src/Controller/AdminController.php

```diff
@@ -186,4 +272,113 @@ private function appendDqlFilterToDqlFilter($dqlFilter, $newDqlFilter)
 
         return $dqlFilter;
     }
+
+    /**
+     * Returns a list of answer-percentages in a period sorted by answer group (1-5).
+     *
+     * @param int       $surveyId
+     * @param \DateTime $fromDate
+     * @param \DateTime $toDate
+     *
+     * @return array
+     *
+     * @throws \Exception
+     */
+    private function getAnswersBetweenDates(int $surveyId, \DateTime $fromDate, \DateTime $toDate): array
+    {
+        $entityManager = $this->getDoctrine()->getEntityManager();
+
+        $query = $entityManager->createQuery('
+            SELECT r.answer,
+                   COUNT(r.answer) as answers
+            FROM App\Entity\Response r
+            WHERE r.survey = :surveyId
+            AND r.createdAt BETWEEN :from AND :to
+            GROUP BY r.answer
+        ');
+
+        $query->setParameter('surveyId', $surveyId);
+        $query->setParameter('from', $fromDate->format('Y-m-d'));
+        $query->setParameter('to', $toDate->add(new \DateInterval('P1D'))->format('Y-m-d'));
+
+        $values = $query->getResult();
+
+        $totalVotes = 0;
+
+        foreach ($values as $value) {
+            $totalVotes += $value['answers'];
+        }
+
+        $newValues = [];
+
+        for ($i = 1; $i < 6; ++$i) {
+            $answers = 0;
+            foreach ($values as $value) {
+                if ($i === $value['answer']) {
+                    $answers = floor((int) $value['answers'] / $totalVotes * 100);
+                }
+            }
+
+            $newValues[] = $answers;
+        }
+
+        return $newValues;
+    }
+
+    /**
+     * Returns the average of answers for a Survey sorted by date in ascending order.
+     * The array returned has to keys, labels which contains an array with the dates that have answers,
+     * and values which contains an array with the average of answers on a date. The first entry in the values
+     * array is the average answers for the first entry in the labels array.
+     *
+     * @param int $surveyId
+     *
+     * @return array
+     */
+    private function getAverageAnswersOnDatesWithLabels(int $surveyId): array
+    {
+        $entityManager = $this->getDoctrine()->getEntityManager();
+
+        $query = $entityManager->createQuery('
+            SELECT r.answer,
+                   COUNT(r.answer) as answers,
+                   DATE(r.createdAt) as dateCreated
+            FROM App\Entity\Response r
+            WHERE r.survey = :surveyId
+            GROUP BY r.answer, dateCreated
+            ORDER BY dateCreated ASC
+        ');
+
+        $query->setParameter('surveyId', $surveyId);
+        $result = $query->getResult();
+
+        $labels = [];
+
+        foreach ($result as $entry) {
+            $labels[] = $entry['dateCreated'];
+        }
+
+        $labels = array_values(array_unique($labels));
+
+        $values = [];
+
+        $totalSumAnswers = 0;
+        $totalSumVotes = 0;
+        foreach ($labels as $date) {
```

Comment:
Add comment describing logic in the this.
Maybe it could be simplified with array_map and array_reduce functions?

Reply:
Commet added by commit: e6ddabb